src/vnet_proxy.erl

```diff
@@ -65,13 +64,15 @@ loop(Pid) ->
   receive Msg -> ?MODULE:loop(Msg, Pid) end.
 
 -spec loop(term(), pid()) -> no_return().
-loop(?noconnection(true), Pid) ->
+loop(?noconnection(From, Ref, true), Pid) ->
   %% Simulate the connection going down
+  From ! Ref,
```

Comment:
Did you consider sending the ack after unlinking, to limit the things that can happen concurrently?